src/NServiceBus.Core/Pipeline/Outgoing/OutgoingPublishContext.cs

```diff
@@ -1,20 +1,21 @@
 ﻿namespace NServiceBus
 {
     using System.Collections.Generic;
+    using Extensibility;
     using Pipeline;
 
     class OutgoingPublishContext : OutgoingContext, IOutgoingPublishContext
     {
-        public OutgoingPublishContext(OutgoingLogicalMessage message, PublishOptions options, IBehaviorContext parentContext)
-            : base(options.MessageId, new Dictionary<string, string>(options.OutgoingHeaders), parentContext)
+        public OutgoingPublishContext(OutgoingLogicalMessage message, string messageId, Dictionary<string,string> headers, ContextBag options, IBehaviorContext parentContext)
```

Comment:
`options` doesn't seem to be the correct name for passing in a `ContextBag`. What about `extensions` as the name instead? Seems to fit with some of the other Context classes.